src/vs/code/electron-main/protocol.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, toDisposable } from 'vs/base/common/lifecycle';
+import { Schemas } from 'vs/base/common/network';
+import { URI } from 'vs/base/common/uri';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import { INativeEnvironmentService } from 'vs/platform/environment/node/environmentService';
+import { session } from 'electron';
+
+export class FileProtocolHandler extends Disposable {
+
+	constructor(
+		@IEnvironmentService private readonly environmentService: INativeEnvironmentService
+	) {
+		super();
+
+		const { defaultSession } = session;
+
+		const resourceHandler = this.handleResourceRequest.bind(this);
+		defaultSession.protocol.registerFileProtocol(Schemas.vscodeFileResource, resourceHandler);
+
+		this._register(toDisposable(() => {
+			defaultSession.protocol.unregisterProtocol(Schemas.vscodeFileResource);
+		}));
+	}
+
+	private async handleResourceRequest(request: Electron.Request, callback: any) {
```

Comment:
Explicit `any` should be avoided

Reply:
The typing for callback is not correct in current `electron.d.ts`https://github.com/electron/electron/blob/master/docs/api/protocol.md#protocolregisterfileprotocolscheme-handler . It should be fixed once we update to electron 11